src/WebJobs.Extensions.DocumentDB/Config/DocumentDBConfiguration.cs

```diff
@@ -56,6 +55,11 @@ internal static string Resolve(string key)
 
                 if (string.IsNullOrEmpty(value))
                 {
+                    value = ConfigurationManager.AppSettings[key];
+                }
+
+                if (string.IsNullOrEmpty(value))
+                {
                     value = Environment.GetEnvironmentVariable(key);
                 }
             }
```

Comment:
@mathewc and I talked about this in another review and punted it for now. The issue was that the default name resolver throws so it wasn't useful and we had to be defensive against that. We decided to open this to track: https://github.com/Azure/azure-webjobs-sdk/issues/653